src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -468,15 +468,16 @@ void IrcMessageHandler::handleClearChatMessage(Communi::IrcMessage *message)
     {
         chan->disableAllMessages();
         chan->addOrReplaceClearChat(std::move(clearChat.message), time);
-        return;
+    }
+    else
+    {
+        chan->addOrReplaceTimeout(std::move(clearChat.message), time);
     }
 
-    chan->addOrReplaceTimeout(std::move(clearChat.message), time);
-
-    // refresh all
-    getApp()->getWindows()->repaintVisibleChatWidgets(chan.get());
     if (getSettings()->hideModerated)
```

Comment:
There should be a check if these are historical messages. In that case we shouldn't force a layout as those messages aren't added to any view yet.

Reply:
As we investigated, this handle function isn't called for historical messages so this is not an issue. Thanks!